src/lib.js

```diff
@@ -214,95 +213,107 @@ export function prosemirrorToYDoc (doc, xmlFragment = 'prosemirror') {
  * collaboration has begun as all history will be lost
  *
  * @param {Schema} schema
- * @param {any} state
+ * @param {any} json
  * @param {string} xmlFragment
  * @return {Y.Doc}
  */
-export function prosemirrorJSONToYDoc (schema, state, xmlFragment = 'prosemirror') {
-  const doc = Node.fromJSON(schema, state)
-  return prosemirrorToYDoc(doc, xmlFragment)
+export function prosemirrorJSONToYDoc (schema, json, xmlFragment = 'prosemirror') {
+  const node = Node.fromJSON(schema, json)
+  return prosemirrorToYDoc(node, xmlFragment)
 }
 
 /**
- * Utility method to convert a Y.Doc to a Prosemirror Doc node.
+ * Utility method to convert a Y.Doc to a Prosemirror node.
  *
  * @param {Schema} schema
  * @param {Y.Doc} ydoc
  * @return {Node}
  */
 export function yDocToProsemirror (schema, ydoc) {
-  const state = yDocToProsemirrorJSON(ydoc)
-  return Node.fromJSON(schema, state)
+  const json = yDocToProsemirrorJSON(ydoc)
+  return Node.fromJSON(schema, json)
 }
 
 /**
  * Utility method to convert a Y.Doc to Prosemirror compatible JSON.
  *
+ * @param {Schema} schema
  * @param {Y.Doc} ydoc
  * @param {string} xmlFragment
  * @return {Record<string, any>}
  */
-export function yDocToProsemirrorJSON (
-  ydoc,
-  xmlFragment = 'prosemirror'
-) {
+export function yDocToProsemirrorJSON (ydoc, xmlFragment = 'prosemirror') {
   const items = ydoc.getXmlFragment(xmlFragment).toArray()
+  return {
+    type: 'doc',
+    content: items.map(typeToProseMirrorJSON).flat()
```

Comment:
The `.flat()` method is a fairly recent addition to EcmaScript. Please use `import { flatten } from 'lib0/array'` instead.